src/MongoDB.Driver.Core/Core/Servers/ServerMonitor.cs

```diff
@@ -61,16 +82,38 @@ public ServerMonitor(ServerId serverId, EndPoint endPoint, IConnectionFactory co
             _baseDescription = _currentDescription = new ServerDescription(_serverId, endPoint, reasonChanged: "InitialDescription", heartbeatInterval: heartbeatInterval);
             _heartbeatInterval = heartbeatInterval;
             _timeout = timeout;
+            _roundTripTimeMonitor = Ensure.IsNotNull(roundTripTimeMonitor, nameof(roundTripTimeMonitor));
+
             _state = new InterlockedInt32(State.Initial);
             eventSubscriber.TryGetEventHandler(out _heartbeatStartedEventHandler);
             eventSubscriber.TryGetEventHandler(out _heartbeatSucceededEventHandler);
             eventSubscriber.TryGetEventHandler(out _heartbeatFailedEventHandler);
             eventSubscriber.TryGetEventHandler(out _sdamInformationEventHandler);
         }
 
-        /// <inheritdoc />
         public ServerDescription Description => Interlocked.CompareExchange(ref _currentDescription, null, null);
 
+        public object Lock => _lock;
+
+        // public methods
+        public void CurrentCheckCancel()
```

Comment:
Nit: naming: `CancelCurrentCheck`?

Reply:
That's unfortunate that the spec uses such a vague term...

Reply:
Resolving for now, feel free to re-open if we'd like to use a name different from the spec!